src/Client/MainComponents/DataMap/DataMap.fs

```diff
@@ -29,13 +31,23 @@ module private Components =
                 let cell = model.SpreadsheetModel.DataMapOrDefault.GetCell(columnIndex, rowIndex)
                 Spreadsheet.Cells.Cell.Body (index, cell, model, dispatch)
                 let isExtended = state.Contains columnIndex
-                if isExtended && (cell.isTerm || cell.isUnitized) then
+                let header = Spreadsheet.Controller.Generic.getHeader columnIndex model.SpreadsheetModel
+                if (cell.isTerm || cell.isUnitized) && isExtended then
                     if cell.isUnitized then 
-                        Spreadsheet.Cells.Cell.BodyUnit(index, cell, model, dispatch)
+                        Cell.BodyUnit(index, cell, model, dispatch)
+                    else
+                        Cell.Empty()
+                    Cell.BodyTSR(index, cell, model, dispatch)
+                    Cell.BodyTAN(index, cell, model, dispatch)
+                elif header.IsDataColumn then
+                    if cell.isData then
+                        Cell.BodyDataSelector(index, cell, model, dispatch)
+                        Cell.BodyDataFormat(index, cell, model, dispatch)
+                        Cell.BodyDataSelectorFormat(index, cell, model, dispatch)
                     else
-                        Spreadsheet.Cells.Cell.Empty()
-                    Spreadsheet.Cells.Cell.BodyTSR(index, cell, model, dispatch)
-                    Spreadsheet.Cells.Cell.BodyTAN(index, cell, model, dispatch)
+                        Cell.Empty()
+                        Cell.Empty()
+                        Cell.Empty()
```

Comment:
MAybe you can replace the three Cell.Empty() method calls witha generic functions that takes an Integer and creates an amount of cells equal to the integer.

Reply:
I mean i could `yield! Array.init 3 (fun _ -> Cell.empty())` but that seems less performant

Reply:
Makes sense